src/engine/GroupBy.cpp

```diff
@@ -961,6 +971,30 @@ void GroupBy::substituteAllAggregates(
   }
 }
 
+// _____________________________________________________________________________
+template <typename A>
+concept SupportedAggregates =
+    ad_utility::SameAsAnyTypeIn<A, GroupBy::AggregationDataVectors>;
+
+template <SupportedAggregates AggregateDataVector>
+struct ResizeVectorsVisitor {
+  void operator()(AggregateDataVector& vector, size_t numberOfGroups,
+                  const GroupBy::HashMapAggregateTypeWithData& info) const {
+    (void)info;
+    vector.resize(numberOfGroups);
+  }
+};
+
+template <>
+struct ResizeVectorsVisitor<std::vector<GroupBy::GroupConcatAggregationData>> {
+  void operator()(std::vector<GroupBy::GroupConcatAggregationData>& vector,
+                  size_t numberOfGroups,
+                  const GroupBy::HashMapAggregateTypeWithData& info) const {
+    vector.resize(numberOfGroups,
+                  GroupBy::GroupConcatAggregationData{info.seperator_.value()});
+  }
```

Comment:
Also for future optimizations, it could be better to store the separator only once, and then maybe only pass a pointer here.